employees/tests/test_unit_report_viewset.py

```diff
@@ -327,6 +327,27 @@ def test_custom_report_list_create_serializer_method_should_return_serializer_wi
             serializer = view._create_serializer()
             self.assertEqual(serializer.fields["date"].initial, "2010-01-21")
 
+    def test_custom_report_list_create_serializer_method_should_return_serializer_filled_with_data_that_was_provided(
+        self
+    ):
+        from django.test import RequestFactory
+
+        request = RequestFactory().post(
+            path=self.url,
+            data={
+                "date": datetime.datetime.now().date(),
+                "description": "Some description",
+                "project": self.project,
+                "work_hours": Decimal("8.00"),
+                "task_activities": TaskActivityType.objects.get(name="Other"),
+            },
+        )
+        request.user = self.user
+        view = ReportList()
+        view.request = request
+        serializer = view._create_serializer(data=request.POST)
+        self.assertTrue(serializer.is_valid())
```

Comment:
Please add test that will prove that for invalid POST request the `project` field value will stay the same.

Reply:
Added this and few more.